tests/integration/targets/aci_vrf_leak_internal_subnet/tasks/main.yml

```diff
@@ -0,0 +1,112 @@
+# Test code for the ACI modules
+# Copyright: (c) 2023, Abraham Mughal (@abmughal)
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Test that we have an ACI APIC host, ACI username and ACI password
+  fail:
+    msg: 'Please define the following variables: aci_hostname, aci_username and aci_password.'
+  when: aci_hostname is not defined or aci_username is not defined or aci_password is not defined
+
+- name: Set vars
+  set_fact:
+    aci_info: &aci_info
+      host: '{{ aci_hostname }}'
+      username: '{{ aci_username }}'
+      password: '{{ aci_password }}'
+      validate_certs: '{{ aci_validate_certs | default(false) }}'
+      use_ssl: '{{ aci_use_ssl | default(true) }}'
+      use_proxy: '{{ aci_use_proxy | default(true) }}'
+      output_level: '{{ aci_output_level | default("debug") }}'
+
+- name: delete existing tenant
+  cisco.aci.aci_tenant: 
+    <<: *aci_info
+    state: absent
+    tenant: ansible_test3
+  register: tenant_present
+
+- name: ensure tenant exists for tests to kick off
+  cisco.aci.aci_tenant: &aci_tenant_present
+    <<: *aci_info
+    state: present
+    tenant: ansible_test3
+  register: tenant_present
+
+- name: create vrf - check mode works
+  cisco.aci.aci_vrf: &aci_vrf_present
+    <<: *aci_tenant_present
+    vrf: test
+    description: Ansible Test
+  check_mode: true
+  register: vrf_present_check_mode
+
+- name: create vrf - creation works
+  cisco.aci.aci_vrf:
+    <<: *aci_vrf_present
+  register: vrf_present
+
+- name: create test - creation works
+  cisco.aci.aci_vrf:
+    <<: *aci_vrf_present
+    vrf: test2
+  register: vrf_present
+
+- name: create test3 - creation works
+  cisco.aci.aci_vrf:
+    <<: *aci_vrf_present
+    vrf: test3
+  register: vrf_present
+
+- name: create leak internal subnet
+  cisco.aci.aci_vrf_leak_internal_subnet: 
+    <<: *aci_info
+    state: present
+    tenant: ansible_test3
+    vrf: test
+    ip: 1.1.1.2
+    leak_internal_subnet: 
+      - vrf: "test2"
+        tenant: "ansible_test3"
+      - vrf: "test3"
+        tenant: "ansible_test3"
+    description: Ansible Test
+  register: vrf_present
+
```

Comment:
Shouldn't you try the check mode on the object you are testing? Instead of the VRF, you should try it on the vrf_leaf_internal_subnet object?